toqito/perms/tests/test_swap.py

```diff
@@ -519,3 +526,10 @@ def test_invalid_swap(input_matrix, sys, dim, row_only):
     """Test function works as expected for an invalid input."""
     with pytest.raises(ValueError):
         swap(input_matrix, sys, dim, row_only)
+
+
+def test_swap_with_invalid_dim_type():
+    """Test swap raises TypeError when dim is of unsupported type."""
+    X = np.arange(1, 17).reshape(4, 4)
+    with pytest.raises(TypeError):
+        swap(X, [1, 2], dim=[{}])
```

Comment:
there is a drop in unit test coverage even though you have a unit test for this line. I wonder if it might not show up as uncovered by unit tests if you verify the message in the unit test.